src/chess/chess_game.rs

```diff
@@ -518,6 +519,96 @@ impl IGame for ChessGame {
     fn get_repetition_limit() -> Option<u32> {
         Some(3)
     }
+
+    fn produce_transformed_data_entries(entry: DataEntry<Self>) -> Vec<DataEntry<Self>> {
+        let transform =
+            |e: &DataEntry<Self>, transform_sq: &dyn Fn(chess::Square) -> chess::Square| {
+                let b = &e.pos.board;
+                let pieces = b
+                    .combined()
+                    .into_iter()
+                    .map(|square| {
+                        (
+                            transform_sq(square),
+                            b.piece_on(square).unwrap(),
+                            b.color_on(square).unwrap(),
+                        )
+                    })
+                    .collect_vec();
+
+                let board = chess::Board::try_from(chess::BoardBuilder::setup(
+                    pieces.iter(),
+                    b.side_to_move(),
+                    b.castle_rights(chess::Color::White),
+                    b.castle_rights(chess::Color::Black),
+                    b.en_passant().map(|square| transform_sq(square).get_file()),
+                ))
+                .expect("unable to transform board");
+                let pos = ChessPosition {
+                    board,
+                    fifty_rule_count: e.pos.fifty_rule_count,
+                };
+
+                let probs = e
+                    .probs
+                    .iter()
+                    .map(|(m, p)| {
+                        (
+                            ChessMove::new(chess::ChessMove::new(
+                                transform_sq(m.m.get_source()),
+                                transform_sq(m.m.get_dest()),
+                                m.m.get_promotion(),
+                            )),
+                            *p,
+                        )
+                    })
+                    .collect_vec();
+
+                let winner = e.winner;
+                DataEntry { pos, probs, winner }
+            };
+
+        let rows_mirror = |e: &DataEntry<Self>| {
+            transform(e, &|sq| {
+                chess::Square::make_square(
+                    chess::Rank::from_index(7 - sq.get_rank().to_index()),
+                    sq.get_file(),
+                )
+            })
+        };
+        let columns_mirror = |e: &DataEntry<Self>| {
+            transform(e, &|sq| {
+                chess::Square::make_square(
+                    sq.get_rank(),
+                    chess::File::from_index(7 - sq.get_file().to_index()),
+                )
+            })
+        };
+        let diagonal_mirror = |e: &DataEntry<Self>| {
+            transform(e, &|sq| {
+                chess::Square::make_square(
+                    chess::Rank::from_index(7 - sq.get_rank().to_index()),
+                    chess::File::from_index(7 - sq.get_file().to_index()),
+                )
+            })
+        };

```

Comment:
one more transform missing? the other diagonal?

Reply:
see other comment